packages/auth-js/src/helpers/authentication-helper.ts

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { KeyLike } from "crypto";
+import axios, { AxiosError, AxiosResponse } from "axios";
+import {
+    AUTHORIZATION_ENDPOINT,
+    CLIENT_ID_TAG,
+    CLIENT_SECRET_TAG,
+    END_SESSION_ENDPOINT,
+    JWKS_ENDPOINT,
+    OIDC_SCOPE,
+    OIDC_SESSION_IFRAME_ENDPOINT,
+    REVOKE_TOKEN_ENDPOINT,
+    SCOPE_TAG,
+    SERVICE_RESOURCES,
+    TOKEN_ENDPOINT,
+    TOKEN_TAG,
+    USERNAME_TAG
+} from "../constants";
+import { DataLayer } from "../data";
+import { AsgardeoAuthException, AsgardeoAuthExceptionStack, AsgardeoAuthNetworkException } from "../exception";
+import { AuthClientConfig, OIDCEndpointsInternal, OIDCProviderMetaData, TokenResponse } from "../models";
+import { AuthenticationUtils, CryptoUtils } from "../utils";
+
+export class AuthenticationHelper<T> {
```

Comment:
IMO, it would be better to document the code. Check other places as well.

Reply:
Created a separate issue to track this: https://github.com/asgardeo/asgardeo-js-oidc-sdk/issues/97